api/v1alpha1/inferencemodel_types.go

```diff
@@ -0,0 +1,167 @@
+	// on other request parameters is to use a different ModelName per HTTPFilter.
+	// Names can be reserved without implementing an actual model in the pool.
+	// This can be done by specifying a target model and setting the weight to zero,
+	// an error will be returned specifying that no valid target model is found.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	ModelName string `json:"modelName,omitempty"`
+	// Defines how important it is to serve the model compared to other models referencing the same pool.
+	//
+	// +optional
+	// +kubebuilder:default="Default"
+	Criticality *Criticality `json:"criticality,omitempty"`
+	// Allow multiple versions of a model for traffic splitting.
+	// If not specified, the target model name is defaulted to the modelName parameter.
+	// modelName is often in reference to a LoRA adapter.
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=10
+	TargetModels []TargetModel `json:"targetModels,omitempty"`
+	// Reference to the poolIt must exist in the same namespace.
+	//
+	// +kubebuilder:validation:Required
+	PoolRef *LocalObjectReference `json:"poolRef,omitempty"`
+}
+
+// LocalObjectReference identifies an API object within the namespace of the
+// referrer.
+type LocalObjectReference struct {
+	// Group is the group of the referent.
+	//
+	// +optional
+	// +kubebuilder:default="inference.networking.x-k8s.io"
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^$|^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+	Group string `json:"group,omitempty"`
+
+	// Kind is kind of the referent. For example "InferencePool".
+	//
+	// +optional
+	// +kubebuilder:default="InferencePool"
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z]([-a-zA-Z0-9]*[a-zA-Z0-9])?$`
+	Kind string `json:"kind,omitempty"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name,omitempty"`
+}
+
+// Defines how important it is to serve the model compared to other models.
+// +kubebuilder:validation:Enum=Critical;Default;Sheddable
+type Criticality string
+
+const (
+	// Most important. Requests to this band will be shed last.
+	Critical Criticality = "Critical"
+	// More important than Sheddable, less important than Critical.
+	// Requests in this band will be shed before critical traffic.
+	// +kubebuilder:default=Default
+	Default Criticality = "Default"
+	// Least important. Requests to this band will be shed before all other bands.
+	Sheddable Criticality = "Sheddable"
+)
+
+// TargetModel represents a deployed model or a LoRA adapter. The
+// Name field is expected to match the name of the LoRA adapter
+// (or base model) as it is registered within the model server. Inference
```

Comment:
I wonder if `TargetModel` is a bit misleading if it also represents an adapter.

Reply:
It does, but this mirrors the Open AI API spec, which just expects a `modelName`. So we chose not to deviate from that pattern.

LMKWYT

Reply:
it may not be necessarily be an adapter too